embedded/sql/stmt.go

```diff
@@ -182,6 +198,14 @@ func (stmt *CreateDatabaseStmt) inferParameters(tx *SQLTx, params map[string]SQL
 }
 
 func (stmt *CreateDatabaseStmt) execAt(tx *SQLTx, params map[string]interface{}) (*SQLTx, error) {
+	if tx.engine.multidbHandler != nil {
+		if tx.explicitClose {
+			return nil, fmt.Errorf("%w: database creation can not be done within a transaction", ErrNonTransactionalStmt)
+		}
+
+		return nil, tx.engine.multidbHandler.CreateDatabase(tx.ctx, stmt.DB)
+	}
+
 	id := uint32(len(tx.catalog.dbsByID) + 1)
 
 	db, err := tx.catalog.newDatabase(id, stmt.DB)
```

Comment:
In case of no `multidbHandler` it will be executed even when we're in the transaction context - is that OK? Maybe we should constraint this too?